src/platform/unittest/PlatformTest.cpp

```diff
@@ -51,3 +51,73 @@ TEST(PlatformTest, QuicAddrParsing)
     }
 }
 
+TEST(PlatformTest, EventQueue)
+{
+    uint32_t user_data1 = 0x1234, user_data2 = 0x5678, user_data3 = 0x90;
+
+    CXPLAT_EVENTQ queue;
+    ASSERT_TRUE(CxPlatEventQInitialize(&queue));
+
+    // Empty queue tests
+    CXPLAT_CQE events[2];
+    ASSERT_EQ(0u, CxPlatEventQDequeue(&queue, events, 2, 0));
+    ASSERT_EQ(0u, CxPlatEventQDequeue(&queue, events, 2, 100));
+
+#ifdef CXPLAT_SQE
+    CXPLAT_SQE sqe1, sqe2, sqe3;
+#ifdef CXPLAT_SQE_INIT
+    ASSERT_TRUE(CxPlatSqeInitialize(&queue, &sqe1, &user_data1));
+    ASSERT_TRUE(CxPlatSqeInitialize(&queue, &sqe2, &user_data2));
+    ASSERT_TRUE(CxPlatSqeInitialize(&queue, &sqe3, &user_data3));
+#endif // CXPLAT_SQE_INIT
+#endif // CXPLAT_SQE
+
+    // Single queue/dequeue tests
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe1, &user_data1));
+    ASSERT_EQ(1u, CxPlatEventQDequeue(&queue, events, 2, 0));
+    ASSERT_EQ((void*)&user_data1, CxPlatCqeUserData(&events[0]));
+
+    // Multiple queue/dequeue tests
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe1, &user_data1));
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe2, &user_data2));
+    ASSERT_TRUE(CxPlatEventQEnqueue(&queue, &sqe3, &user_data3));
+    ASSERT_EQ(2u, CxPlatEventQDequeue(&queue, events, 2, 100));
```

Comment:
why wait 100ms for the first 2 completions, but 0 ms for the third completion?

Reply:
Just for the sake of changing things up. No specific reason.